static/js/components/EducationDialog.js

```diff
@@ -23,11 +26,16 @@ export default class EducationDialog extends ProfileFormFields {
   }
   educationLevelLabels: Object;
 
-  static propTypes = {
-    open:           React.PropTypes.bool,
-    onClose:        React.PropTypes.func,
-    onSave:         React.PropTypes.func,
-    showLevelForm:  React.PropTypes.bool,
+  props: {
+    ui:                           UIState,
+    setEducationDialogVisibility: () => void,
+    setEducationDegreeLevel:      () => void,
+    setEducationDialogIndex:      () => void,
+    clearProfileEdit:             () => void,
+    saveProfile:                  BoundSaveProfile,
```

Comment:
it's a declared type because we need to annotate it in several places, flow enforces type declarations at module boundaries so there's a little bit of redundancy there. So we need to have the declared type (it's declared in `flow/profileTypes.js`) in order to share the annotation (and not have `(validator: Validator|UIValidator, profile: Profile, ui: UIState) => Promise<Profile>`, which is a bit verbose, all over the place).


I called it `BoundSaveProfile` because it's the result of taking the `saveProfile` method on `ProfileFormContainer` and binding `this` and the first positional argument (`isEdit`).